algorithms/ar-expsq/golang/exponentiation_by_square.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import "fmt"
+
+var mod int64 = 1000000007
+
+func main() {
+	var base int64 = 5
+	var exponent int64 = 100000
+	var value int64 = exponentiationBySquaring(base, exponent)
```

Comment:
Codacy found an issue: [should omit type int64 from declaration of var value; it will be inferred from the right-hand side](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6393195)

Reply:
Done. Please review